packagedb/models.py

```diff
@@ -1219,6 +1219,153 @@ def make_relationship(
         relationship=relationship,
     )
 
+class UpdateMixin:
+    """
+    Provide a ``update()`` method to trigger a save() on the object with the
+    ``update_fields`` automatically set to force a SQL UPDATE.
+    """
+
+    def update(self, **kwargs):
+        """
+        Update this resource with the provided ``kwargs`` values.
+        The full ``save()`` process will be triggered, including signals, and the
+        ``update_fields`` is automatically set.
+        """
+        for field_name, value in kwargs.items():
+            setattr(self, field_name, value)
+
+        self.save(update_fields=list(kwargs.keys()))
+
+
+class PackageWatch(models.Model, UpdateMixin):
+    """
+    Model representing a watch on a package to monitor for new versions.
+    """
+
+    DEPTH_CHOICES = (
+        ("version", "Version"),
```

Comment:
Use a numbers instead: 1,2,3 and the field can then be a PositiveSmallIntegerField for this. See IntegerChoices class too.